vscode-extensions/snippets/snippets/snippets.code-snippets

```diff
@@ -0,0 +1,803 @@
+      "    Content",
+      "  </div>}",
+      "</ef-collapse>",
+    ]
+  },
+  "Collapse: Levels": {
+    "prefix": "ef-collapse:level",
+    "body": [
+      "<ef-collapse level=\"1\" header=\"${1:Level 1}\">",
+      "  ${2:<div>",
+      "    Content",
+      "  </div>}",
+      "</ef-collapse>",
+      "<ef-collapse level=\"2\" header=\"${3:Level 2}\">",
+      "  ${4:<div>",
+      "    Content",
+      "  </div>}",
+      "</ef-collapse>",
+      "<ef-collapse level=\"3\" header=\"${5:Level 3}\">",
+      "  ${6:<div>",
+      "    Content",
+      "  </div>}",
+      "</ef-collapse>",
+    ]
+  },
+  "Color Dialog": {
+    "prefix": "ef-color-dialog",
+    "body": [
+      "<ef-color-dialog id=\"${1:efColorDialog}\" ${2:allow-nocolor} ${3:value=\"$4\"}></ef-color-dialog>"
+    ]
+  },
+  "Color Picker": {
+    "prefix": "ef-color-picker",
+    "body": [
+      "<ef-color-picker id=\"${1:efColorPicker}\" ${2:allow-nocolor} ${3:value=\"$4\"}></ef-color-picker>"
+    ]
+  },
+  "Combo Box": {
+    "prefix": "ef-combo-box",
+    "body": [
+      "<ef-combo-box id=\"${1:efComboBox}\" ${2:multiple} ${2:data=\"$3\"} ${4:value=\"$5\"}></ef-combo-box>"
+    ]
+  },
+  "Combo Box: Example": {
+    "prefix": "ef-combo-box@example",
+    "body": [
+      "<ef-combo-box id=\"${1:efComboBox}\" ${2:multiple}></ef-combo-box>",
+      "<script>",
+      "  const comboBox = document.querySelector('${1:efComboBox}');",
+      "  comboBox.data = [",
+      "    { label: 'EMEA', type: 'header' },",
+      "    { label: 'France', value: 'fr' },",
+      "    { label: 'United Kingdom', value: 'gb', selected: true }",
+      "  ];",
+      "</script>"
+    ]
+  },
+  "Counter": {
+    "prefix": "ef-counter",
+    "body": [
+      "<ef-counter ${1:max=\"$2\"} ${3:value=\"$4\"}></ef-counter>"
+    ]
+  },
+  "Datetime Picker": {
+    "prefix": "ef-datetime-picker",
+    "body": [
+      "<ef-datetime-picker id=\"${1:efDatetimePicker}\" ${2:timepicker} ${3:value=\"$4\"}></ef-datetime-picker>"
+    ]
+  },
+  "Datetime Picker: Range": {
+    "prefix": "ef-datetime-picker:range",
+    "body": [
+      "<ef-datetime-picker id=\"${1:efDatetimePicker}\" ${2:range} ${3:duplex=\"${4:split}\"} ${5:values=\"${6:$CURRENT_YEAR-$CURRENT_MONTH-$CURRENT_DATE},${7:$CURRENT_YEAR-$CURRENT_MONTH-$CURRENT_DATE}\"}></ef-datetime-picker>"
+    ]
+  },
+  "Dialog": {
+    "prefix": "ef-dialog",
+    "body": [
+      "<ef-dialog id=\"${1:efDialog}\" header=\"${2:Title}\">",
+      "  ${3:<p>Title</p>}",
+      "</ef-dialog>",
+    ]
+  },
+  "Dialog: Footer": {
+    "prefix": "ef-dialog:footer-slot",
+    "body": [
+      "<ef-dialog id=\"${1:efDialog}\" header=\"${2:Title}\">",
+      "  ${3:<p>Title</p>}",
+      "  <div slot=\"footer\">",
+      "     $4",
+      "  </div>",
+      "</ef-dialog>",
+    ]
+  },
+  "Email Field": {
+    "prefix": "ef-email-field",
+    "body": [
+      "<ef-email-field id=\"${1:efEmailField}\" ${2:value=\"$3\"}></ef-email-field>"
```

Comment:
I think we should not duplicate the icon list in many places for now because of maintenance reasons. we can do it when have the scripts to inject the icon list into the all icon attribute in the snippet file. 😀